src/libs/SidebarUtils.js

```diff
@@ -151,9 +151,23 @@ function getOrderedReportIDs(reportIDFromRoute) {
 
     // Sort each group of reports accordingly
     pinnedReports = _.sortBy(pinnedReports, report => report.displayName.toLowerCase());
-    outstandingIOUReports = _.sortBy(outstandingIOUReports, 'iouReportAmount').reverse();
+    outstandingIOUReports = outstandingIOUReports.sort((a, b) => {
+        if (a.iouReportAmount > b.iouReportAmount) {
+            return -1;
+        }
+        if (a.iouReportAmount < b.iouReportAmount) {
+            return 1;
+        }
+        return a.displayName.toLowerCase().localeCompare(b.displayName.toLowerCase());
+    });
     draftReports = _.sortBy(draftReports, report => report.displayName.toLowerCase());
-    nonArchivedReports = _.sortBy(nonArchivedReports, report => (isInDefaultMode ? report.lastMessageTimestamp : report.displayName.toLowerCase()));
+    nonArchivedReports = nonArchivedReports.sort((a, b) => {
```

Comment:
i noticed that nonArchivedReports also showed up very differently between web and mobile. I made some improvements so that it sorts more consistently between mobile and web. 

But its not perfect. There is a chance both messageTimestamp and displayNames are the same. For example i noticed for some #admin rooms the order would be off. We could further sort on policyID but i didn't want to go into a rabbit hole if it's not currently a problem and I'm content with the current improvement.